File: app/src/main/java/com/erotc/learning/data/QuestionEntry.java
package com.erotc.learning.data;

import com.j256.ormlite.field.DatabaseField;
import com.j256.ormlite.table.DatabaseTable;

/**
 * Created on 10/15/2018.
 */
@DatabaseTable
public class QuestionEntry {
    @DatabaseField(generatedId = true)
    private int id;

    @DatabaseField(foreign = true, foreignAutoRefresh = true)
    private QuestionSet questionSet;

    @DatabaseField
    private String question;

    @DatabaseField
    private String answer;

    public QuestionEntry() {
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public QuestionSet getQuestionSet() {
        return questionSet;
    }

    public void setQuestionSet(QuestionSet questionSet) {
        this.questionSet = questionSet;
    }

    public String getQuestion() {
        return question;
    }

    public void setQuestion(String question) {
        this.question = question;
    }

    public String getAnswer() {
        return answer;
    }

    public void setAnswer(String answer) {
        this.answer = answer;
    }
}

File: app/src/main/java/com/erotc/learning/data/QuestionEntry.kt
package com.erotc.learning.data

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.PrimaryKey

/**
 * Created on 10/15/2018.
 */
@Entity(foreignKeys = [
    ForeignKey (
        entity = QuestionSet::class,
        parentColumns = arrayOf("id"),
        childColumns = arrayOf("questionSetId"),
        onDelete = ForeignKey.CASCADE
    )
])
data class QuestionEntry (
    @PrimaryKey(autoGenerate = true) var id: Long = 0,
    var questionSetId: Long = 0,
    var question: String? = null,
    var answer: String? = null
)